test/unit/deployment.bats

```diff
@@ -178,6 +178,75 @@ load _helpers
     [ "${actual}" = "true" ]
 }
 
+# podSecurityContext
+@test "controller/Deployment: controller.podSecurityContext set by default" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.securityContext | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+   [ "${actual}" = "1" ]
+   actual=$(echo "$object" | yq '.runAsNonRoot' | tee /dev/stderr)
+   [ "${actual}" = "true" ]
+
+}
+
+@test "controller/Deployment: controller.podSecurityContext can be set" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.podSecurityContext.runAsGroup=2000' \
+      --set 'controller.podSecurityContext.runAsUser=2000' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.securityContext | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+   [ "${actual}" = "2" ]
```

Comment:
Test failure here:

```
not ok 38 controller/Deployment: controller.podSecurityContext can be set
# (in test file test/unit/deployment.bats, line 206)
#   `[ "${actual}" = "2" ]' failed
# 3
```

I think we can just change the expectation to 3, as `runAsNonRoot` should still be set with the way the test is set up.

```suggestion
   [ "${actual}" = "3" ]
```